### PR TITLE
implement cryptocom

### PR DESCRIPTION
**Pull request by frosty00**
cleaned and tested

### CODE REVIEW

File: js/cryptocom.js [hunk 1]
```diff
@@ -0,0 +1,1207 @@
...
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'] + '/' + path;
+        const query = this.omit (params, this.extractParams (path));
+        if (api === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const id = '1';
```
Comment (kroitor) [inline js/cryptocom.js:221]: Setting this id to a constant value isn't really correct.
We have to use `this.milliseconds()` or something similar here.
Their docs say it must be a long value, not a string.

File: js/cryptocom.js [hunk 2]
```diff
@@ -0,0 +1,1207 @@
...
+        //     }
+        // }
+        const resultResponse = this.safeValue (response, 'result', {});
+        const data = this.safeValue (resultResponse, 'data', {});
+        return this.parseTicker (data, market);
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
```
Comment (kroitor) [inline js/cryptocom.js:378]: Let's move this symbol-required warning above `loadMarkets()` everywhere.

File: js/cryptocom.js [hunk 3]
```diff
@@ -0,0 +1,1207 @@
...
+            account['free'] = this.safeString (balance, 'available');
+            account['used'] = this.safeString (balance, 'order');
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
```
Comment (kroitor) [inline js/cryptocom.js:555]: The symbol is `undefined` by default.

File: js/cryptocom.js [hunk 4]
```diff
@@ -0,0 +1,1207 @@
...
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const uppercaseType = type.toUpperCase ();
+        const request = {
+            'instrument_name': market['id'],
+            'side': side.toUpperCase (),
+            'type': type.toUpperCase (),
```
Comment (kroitor) [inline js/cryptocom.js:608]: No need to uppercase the type twice.

File: js/cryptocom.js [hunk 5]
```diff
@@ -0,0 +1,1207 @@
...
+        // }
+        const data = this.safeValue (response, 'result', {});
+        const value = this.safeValue (data, 'order_id');
+        const order = await this.fetchOrder (value, symbol);
+        return order;
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
```
Comment (kroitor) [inline js/cryptocom.js:636]: This warning should throw early, before loadMarkets().

File: js/cryptocom.js [hunk 6]
```diff
@@ -0,0 +1,1207 @@
...
+        const market = this.market (symbol);
+        const request = {
+            'instrument_name': market['id'],
+        };
+        return await this.privatePostPrivateCancelAllOrders (this.extend (request, params));
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
```
Comment (kroitor) [inline js/cryptocom.js:648]: Let's move this above loadMarkets().

File: js/cryptocom.js [hunk 7]
```diff
@@ -0,0 +1,1207 @@
...
+                'fetchTickers': true,
+                'fetchTime': false,
+                'fetchTrades': true,
+                'fetchTradingFee': false,
+                'fetchTradingFees': false,
+                'fetchTransactions': false,
+                'fetchWithdrawals': false,
+                'setLeverage': false,
+                'setMarginMode': false,
+                'withdraw': false,
```
Comment (kroitor) [inline js/cryptocom.js:52]: Looks like they have a `withdraw` method: https://exchange-docs.crypto.com/spot/index.html#private-create-withdrawal

Comment (kroitor) [reply js/cryptocom.js:52]: They also have fetchDeposits, fetchWithdrawals and transfer.

Comment (frosty00) [reply js/cryptocom.js:52]: yeah...

Comment (frosty00) [reply js/cryptocom.js]: I think we can merge it for now without transfer and then add it once we add support for margin and derivatives (currently it only supports spot)

Comment (kroitor) [reply js/cryptocom.js]: We should still add depositing and withdrawing though.

Comment (kroitor) [reply js/cryptocom.js]: Ah, nvm, now i see them added.

File: js/cryptocom.js [hunk 8]
```diff
@@ -0,0 +1,1210 @@
...
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrder requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'order_id': id,
```
Comment (kroitor) [inline js/cryptocom.js:560]: I think we're missing a market id in this request, otherwise I'm not entirely sure why we would require a `symbol`, right?

**Comment by frosty00** [comment]
i've made a few edits, should be good to merge now

**Pull request merged**